src/utils/parseUrl.js

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import parseUri from "parse-uri";
+import isString from "./isString";
+
+const parseDomainBasic = host => {
+  const result = {};
+  const domainParts = host.split(".");
+
+  switch (domainParts.length) {
+    case 1:
+      result.subdomain = "";
+      result.domain = host;
+      result.topLevelDomain = "";
+      break;
+    case 2:
+      result.subdomain = "";
+      result.domain = host;
+      result.topLevelDomain = domainParts[1];
+      break;
+    case 3:
+      result.subdomain = domainParts[0] === "www" ? "" : domainParts[0];
+      result.domain = host;
+      result.topLevelDomain = domainParts[2];
+      break;
+    case 4:
+      result.subdomain = domainParts[0] === "www" ? "" : domainParts[0];
+      result.domain = host;
+      result.topLevelDomain = `${domainParts[2]}.${domainParts[3]}`;
+      break;
```

Comment:
The topLevelDomain fails for country-code top level domains i.e. "amazon.co.uk"

Reply:
I verified with a unit test for amazon.co.uk. It passes 
```
     it("should parse a URL with country-code top level components", () => {
    const url = "amazon.co.uk";
    const result = parseUrl(url);
    expect(result.topLevelDomain).toBe("uk");
  });
```
[ country code top level domain](https://en.wikipedia.org/wiki/Country_code_top-level_domain) is uk